harvest/trader/trader.py

```diff
@@ -171,6 +168,11 @@ async def timeout(self):
         except asyncio.CancelledError:
             debug("Timeout cancelled")
 
+    async def run_on_interval(func, interval, *args, **kwargs):
+        while True:
+            await asyncio.sleep(interval)
+            func(*args, **kwargs)
```

Comment:
To be used in the future to run things like the streamer main or the algo main on an interval instead of calling the streamer which calls the trader which calls the algo.